node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -505,3 +506,571 @@ where
         COMPONENT_NAME
     }
 }
+
+#[cfg(test)]
```

Comment:
Nit:

Because there's quite a lot of tests, consider moving them to a dedicated module in `block_synchronizer/tests.rs`.

Reply:
Move both trie accumulator and global state synchronizer tests in separate files in https://github.com/casper-network/casper-node/pull/3568/commits/9cf57e53b38f6c80dea20521504ba8b3b8c93ce8

Reply:
I think it would be more idiomatic to organize the tests like so:
```
├── block_synchronizer
│   ├── global_state_synchronizer
│   │   └── tests.rs
│   ├── global_state_synchronizer.rs
│   ├── trie_accumulator
│   │   └── tests.rs
│   └── trie_accumulator.rs
```

That'll allow removing the explicit [`path`](https://github.com/casper-network/casper-node/pull/3568/files#diff-66baebf8cc9d7747b8595b3dfa149423db0265d8bc3ce43e975892c0c12154e6R2) attribute.

Wdyt?

Compare to, for example, `node/src/components/consensus/highway_core/state/tests.rs`